Using a dropper to launch slime blocks into a water-filled cauldron will result in a game crash.

Minecraft version:1.19.2
Spelunkery version:0.2.5
Moonlight Lib version:2.2.44
Fabric API version: 0.76.1


Logs:
[17:03:23] [Server thread/ERROR]: Encountered an unexpected exception
net.minecraft.class_148: Colliding entity with block
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:879) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:819) ~[client-intermediary.jar:?]
	at net.minecraft.class_1132.method_3748(class_1132.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:665) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:257) ~[client-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "net.minecraft.class_3222.method_14220()" because "$$0" is null
	at net.minecraft.class_4711.method_23889(class_4711.java:33) ~[client-intermediary.jar:?]
	at net.minecraft.class_5556.handler$cpm000$spelunkery$entityInside(class_5556.java:556) ~[client-intermediary.jar:?]
	at net.minecraft.class_5556.method_9548(class_5556.java) ~[client-intermediary.jar:?]
	at net.minecraft.class_4970$class_4971.method_26178(class_4970.java:952) ~[client-intermediary.jar:?]
	at net.minecraft.class_1297.method_5852(class_1297.java:1059) ~[client-intermediary.jar:?]
	at net.minecraft.class_1297.method_36974(class_1297.java:783) ~[client-intermediary.jar:?]
	at net.minecraft.class_1297.method_5784(class_1297.java:752) ~[client-intermediary.jar:?]
	at net.minecraft.class_1542.method_5773(class_1542.java:132) ~[client-intermediary.jar:?]
	at net.minecraft.class_3218.method_18762(class_3218.java:712) ~[client-intermediary.jar:?]
	at net.minecraft.class_1937.method_18472(class_1937.java:466) ~[client-intermediary.jar:?]
	at net.minecraft.class_3218.method_31420(class_3218.java:371) ~[client-intermediary.jar:?]
	at net.minecraft.class_5574.method_31791(class_5574.java:54) ~[client-intermediary.jar:?]
	at net.minecraft.class_3218.method_18765(class_3218.java:339) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:875) ~[client-intermediary.jar:?]
	... 5 more

--- a/common/src/main/java/com/ordana/spelunkery/mixins/WaterCauldronBlockMixin.java
+++ b/common/src/main/java/com/ordana/spelunkery/mixins/WaterCauldronBlockMixin.java
@@ -43,18 +43,23 @@ public class WaterCauldronBlockMixin extends AbstractCauldronBlock {
     @Inject(method = "entityInside", at = @At("HEAD"))
     public void entityInside(BlockState state, Level level, BlockPos pos, Entity entity, CallbackInfo ci) {
         if (entity instanceof ItemEntity item && state.is(Blocks.WATER_CAULDRON) && this.isEntityInsideContent(state, pos, entity) && level instanceof ServerLevel serverLevel) {
-            ItemStack itemStack2 = item.getItem();
-            if (itemStack2.is(ModItems.SALT.get()) && serverLevel.getBlockState(pos.below()).is(ModTags.CAN_BOIL_WATER)) {
-                int count = itemStack2.getCount();
-                ItemStack itemStack = new ItemStack(ModItems.ROCK_SALT.get());
-                itemStack.setCount(count);
-                item.setItem(itemStack);
+            ItemStack itemStack = item.getItem();
+            if (itemStack.is(ModItems.SALT.get()) && serverLevel.getBlockState(pos.below()).is(ModTags.CAN_BOIL_WATER)) {
+                int count = itemStack.getCount();
+                ItemStack newStack = new ItemStack(ModItems.ROCK_SALT.get());
+                newStack.setCount(count);
+                item.setItem(newStack);
                 LayeredCauldronBlock.lowerFillLevel(state, serverLevel, pos);
             }
-            if (itemStack2.is(Items.SLIME_BLOCK) && CommonConfigs.SLIME_CAULDRONS.get() && state.getValue(LEVEL) == 3) {
+            if (itemStack.is(Items.SLIME_BLOCK) && CommonConfigs.SLIME_CAULDRONS.get() && state.getValue(LEVEL) == 3) {
                 Slime slime = EntityType.SLIME.create(serverLevel);
                 if (slime != null) {
-                    CriteriaTriggers.ITEM_USED_ON_BLOCK.trigger((ServerPlayer) Objects.requireNonNull(item.getOwner()), pos, itemStack2);
+                    try {
+                        ServerPlayer player = (ServerPlayer) item.getOwner();
+                        CriteriaTriggers.ITEM_USED_ON_BLOCK.trigger(player, pos, itemStack);
+                    } catch (NullPointerException|ClassCastException ex) {
+                        // ignored -- not thrown by a player
+                    }
                     item.remove(Entity.RemovalReason.DISCARDED);
                     slime.moveTo(pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 0.5);
                     serverLevel.addFreshEntity(slime);

--- a/common/src/main/java/com/ordana/spelunkery/mixins/WaterCauldronBlockMixin.java
+++ b/common/src/main/java/com/ordana/spelunkery/mixins/WaterCauldronBlockMixin.java
@@ -56,7 +56,7 @@ public class WaterCauldronBlockMixin extends AbstractCauldronBlock {
                 if (slime != null) {
                     try {
                         ServerPlayer player = (ServerPlayer) item.getOwner();
-                        CriteriaTriggers.ITEM_USED_ON_BLOCK.trigger(player, pos, itemStack);
+                        CriteriaTriggers.ITEM_USED_ON_BLOCK.trigger(Objects.requireNonNull(player), pos, itemStack);
                     } catch (NullPointerException|ClassCastException ex) {
                         // ignored -- not thrown by a player
                     }

